LanguageFeatures/Patterns/variable_A02_t05.dart

```diff
@@ -45,8 +45,37 @@ String test(Map m) {
   }
 }
 
+String test2(Map m) {
+  return switch (m) {
+    case ({1: String b}) => "{1: <String>$a}";
+    case ({1: final int b}) => "{1: <int>$b}";
+    default => "default";
+  };
+}
+
+String test3(Map m) {
+  if (m case ({1: String b})) {
+    a.expectStaticType<Exactly<String>>();
```

Comment:
Ah, this one would work! I might have overlooked that some of the others would work as well, because there is a declared type directly on the variable.